src/Sylius/Component/Core/Factory/PaymentMethodFactory.php

```diff
@@ -54,8 +55,10 @@ public function createNew(): PaymentMethodInterface
      */
     public function createWithGateway(string $gatewayFactory): PaymentMethodInterface
     {
+        /** @var GatewayConfigInterface $gatewayConfig */
         $gatewayConfig = $this->gatewayConfigFactory->createNew();
         $gatewayConfig->setFactoryName($gatewayFactory);
+        $gatewayConfig->setGatewayName(ucfirst($gatewayFactory));
```

Comment:
If I think correctly, the `gatewayName` have to be unique in Payum, so that is probably not the best solution /cc @Arminek 

Now, the gateway name is set as `code` of `PaymentMethod` and this change could change current behaviour.
https://github.com/Sylius/Sylius/blob/49a1ac51277e5ba98b45db7e2d4f2bfad1155e71/src/Sylius/Bundle/CoreBundle/Form/Extension/PaymentMethodTypeExtension.php#L55-L58